### ISSUE TITLE
ARM template deployment to NE fails with 'Operation Expired (Code 90008)'

### ISSUE THREAD

**Issue by adrianbontea** [microsoft/azure-container-apps#84]
Hello,

I am trying to deploy one Azure Container App for a basic REST API component as part of a POC to North Europe region. I am using an ARM template. The deployment always fails with 'Operation Expired (Code 90008)' regardless if I use a PS script or using the template manually from Azure Portal. My template:

```
{
  "$schema": "https://schema.management.azure.com/schemas/2019-08-01/deploymentTemplate.json#",
  "contentVersion": "1.0.0.0",
  "parameters": {
    "location": {
      "defaultValue": "northeurope",
      "type": "String"
    },
    "environment_name": {
      "type": "String"
    },
    "cosmos_account_url": {
      "type": "String"
    },
    "cosmos_account_master_key": {
      "type": "String"
    },
    "cosmos_account_database": {
      "type": "String"
    },
    "cosmos_account_container": {
      "type": "String"
    },
    "service_bus_connection_string": {
      "type": "String"
    },
    "acr": {
      "type": "String"
    },
    "acr_username": {
      "type": "String"
    },
    "acr_password": {
      "type": "String"
    }
  },
  "variables": {
    "pubSubComponentName": "weather-forecast-pub-sub",
    "pubSubTopicName": "weather-forecast-request",
    "stateComponentName": "weather-forecast-state",
    "acrUrl": "[concat(parameters('acr'),'.azurecr.io')]"
  },
  "resources": [
    {
      "name": "weather-forecast-api",
      "type": "Microsoft.Web/containerApps",
      "apiVersion": "2021-03-01",
      "kind": "containerapp",
      "location": "[parameters('location')]",
      "properties": {
        "kubeEnvironmentId": "[resourceId('Microsoft.Web/kubeEnvironments', parameters('environment_name'))]",
        "configuration": {
          "ingress": {
            "external": true,
            "targetPort": 8080
          },
          "secrets": [
            {
              "name": "cosmos-account-master-key",
              "value": "[parameters('cosmos_account_master_key')]"
            },
            {
              "name": "service-bus-connection-string",
              "value": "[parameters('service_bus_connection_string')]"
            },
            {
              "name": "acr-password",
              "value": "[parameters('acr_password')]"
            }
          ],
          "registries": [
            {
              "server": "[variables('acrUrl')]",
              "username": "[parameters('acr_username')]",
              "passwordSecretRef": "acr-password"
            }
          ]
        },
        "template": {
          "containers": [
            {
              "image": "[concat(variables('acrUrl'),'/weather-forecast/weather-forecast-api:latest')]",
              "name": "weather-forecast-api",
              "env": [
                {
                  "name": "Dapr_PubSubComponentName",
                  "value": "[variables('pubSubComponentName')]"
                },
                {
                  "name": "Dapr_WeatherForecastTopicName",
                  "value": "[variables('pubSubTopicName')]"
                },
                {
                  "name": "Dapr_StateComponentName",
                  "value": "[variables('stateComponentName')]"
                }
              ],
              "resources": {
                "cpu": 0.5,
                "memory": "1Gi"
              }
            }
          ],
          "scale": {
            "minReplicas": 1,
            "maxReplicas": 1
          },
          "dapr": {
            "enabled": true,
            "appPort": 50001,
            "appId": "weather-forecast-api",
            "components": [
              {
                "name": "[variables('stateComponentName')]",
                "type": "state.azure.cosmosdb",
                "version": "v1",
                "metadata": [
                  {
                    "name": "url",
                    "value": "[parameters('cosmos_account_url')]"
                  },
                  {
                    "name": "masterKey",
                    "secretRef": "cosmos-account-master-key"
                  },
                  {
                    "name": "database",
                    "value": "[parameters('cosmos_account_database')]"
                  },
                  {
                    "name": "collection",
                    "value": "[parameters('cosmos_account_container')]"
                  }
                ]
              },
              {
                "name": "[variables('pubSubComponentName')]",
                "type": "pubsub.azure.servicebus",
                "version": "v1",
                "metadata": [
                  {
                    "name": "connectionString",
                    "secretRef": "service-bus-connection-string"
                  }
                ]
              }
            ]
          }
        }
      }
    }
  ]
}
```
The image `'/weather-forecast/weather-forecast-api:latest` has been correctly built and pushed to ACR. The implementation is a basic ASP.NET Core 6 Web API (TF .net 6), based on the MS Weather Forecast sample with minimal enhancements using Dapr ASB pub/sub component on a POST endpoint and Dapr CosmosDB state store component on a GET endpoint.

All required Azure resources (Container Apps Environment, Log Analytics Workspace, Container Registry, Cosmos DB Account, Service Bus Namespace) have been correctly provisioned in the RG I am trying to deploy.

**Comment by adrianbontea** [comment]
Same error if I just try to create the container app manually, from Azure Portal, using ACR and the same image from the template above (previously pushed to ACR) and no Dapr configuration:

`{"code":"DeploymentFailed","message":"At least one resource deployment operation failed. Please list deployment operations for details. Please see https://aka.ms/DeployOperations for usage details.","details":[{"code":"90008","message":"Failed to provision revision for container app 'weather-forecast-api'. Error details: Operation expired."}]}`

**Comment by ahmelsayed** [comment]
It seems that you have another app with a dapr component named `weather-forecast-state`. An app called `weather-forecast-process`. There is an enforcement that component names should be unique across the environment, but we're pushing a fix for that. There is another reporting issue where that error is not making it back to the response. That is also tracked. 

We started a deployment with a fix for the first issue, and I can update here once it's deployed to NE. Or you can try changing the component name, but I know that might be inconvenient because you'll have to change it in code as well.

**Comment by adrianbontea** [comment]
Thanks @ahmelsayed, that's right, I have another CA deployed to the same CAE using Dapr state and pub/sub components with the same names. That deployed correctly via the ARM template. Please update this issue once the fix for unique component names has been deployed to NE. In the meantime, I will also try to deploy my API CA with different names for the Dapr components

**Comment by adrianbontea** [comment]
I was able to successfully deploy from the ARM template above after changing the Dapr components name to be unique across the Container Apps Environment. Didn't have to change any code in the API component implementation (except ARM) since the Dapr pub/sub and state component names are configurable, injected into the container as environment variables from the variables in the ARM template. Closing this issue now since the other two issues mentioned by @ahmelsayed are being tracked separately.
